### PR TITLE
[JENKINS-46523] - Make UnixReflection class compatible with Java 11, remove pre-Java 8 reflection

### PR DESCRIPTION
**Pull request by oleg-nenashev**
See [JENKINS-46523](https://issues.jenkins-ci.org/browse/JENKINS-46523). UnixReflection class didn't support Java 9+ APIs, and `java.lang.UNIXProcess` is removed from recent Java versions. Multi-release JAR is probably a way to go, but so far I would like to apply a quick patch.

This patch fixes ~100 tests in #3636 

### Proposed changelog entries

* RFE: Update Unix process management logic to support Process Tree termination in Java 11
* ...

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-platform

### CODE REVIEW

File: core/src/main/java/hudson/util/ProcessTree.java [hunk 1]
```diff
@@ -818,60 +812,94 @@ private void killRecursively(long deadline) throws InterruptedException {
...
+
+        //TODO: We ideally need to update ProcessTree APIs to Support Long (JENKINS-53799).
+        public static int pid(@Nonnull Process proc) {
+            try {
+                if (JAVA8_PID_FIELD != null) {
+                    return JAVA8_PID_FIELD.getInt(proc);
+                } else {
+                    long pid = (long)JAVA9_PID_METHOD.invoke(proc);
+                    if (pid > Integer.MAX_VALUE) {
+                        throw new IllegalAccessError("Java 9+ support error: PID is out of Jenkins API bounds: " + pid);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/util/ProcessTree.java]: Probably I should link JENKINS-53799 explicitly from here

**Review by MarkEWaite — APPROVED** [review]

**Comment by Mr-jiangzhiguo** [comment]
@oleg-nenashev @MarkEWaite 
Excuse me, I have a problem: 
 jenkins: `v2.138.1`
ssh-plugin: `v2.6.1`
the "`Add Credentials`" button don't work, how to do???

Otherwise, I have only one server, I wanna deploy a nodejs project though jenkins with docker on it,
so, what should I do?

**Comment by MarkEWaite** [comment]
Pull requests are not the place for unrelated technical questions. Ask elsewhere

**Comment by Mr-jiangzhiguo** [comment]
> Pull requests are not the place for unrelated technical questions. Ask elsewhere

ok,  but there is no issuse entry

**Comment by oleg-nenashev** [comment]
* https://jenkins.io/chat/
* https://groups.google.com/forum/#!forum/jenkinsci-users
* https://wiki.jenkins.io/display/JENKINS/How+to+report+an+issue

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**